aten/src/ATen/native/cuda/UnaryFractionKernels.cu

```diff
@@ -69,10 +69,18 @@ __host__ __device__ static inline thrust::complex<T> reciprocal_wrapper(thrust::
     return (::isinf(real) && ::isinf(imag));
   };
 
-  if (::isnan(v.real()) || ::isnan(v.imag()) || both_inf(v.real(), v.imag())) {
+  auto either_inf = [](T real, T imag) {
```

Comment:
Should comment saying why! And what version of CUDA the bug was

Reply:
I believe it was 10.1.168 on x64-linux